src/test/java/ControllerTest.java

```
@@ -20,66 +19,167 @@
+    private Filter<Product> atLeastTwoFilter;
+
+    private Product blackDogFor500;
+    private Product blueCatFor1010;
+    private Product yellowMouseFor840;
+    private Product blackCatFor1001;
+
+    private List<Product> products;
 
     @Before
     public void beforeMethod() {
         input = Mockito.mock(Input.class);
         output = Mockito.mock(Output.class);
         logger = Mockito.mock(Logger.class);
+        products = new ArrayList<>();
 
-        blackDog = new Product(0, "Dog", Color.BLACK, BigDecimal.valueOf(500));
-        blackCat = new Product(0, "Cat Black", Color.BLACK, BigDecimal.valueOf(1000));
-        blueCat = new Product(0, "Cat", Color.BLUE, BigDecimal.valueOf(1000));
-        yellowMouse = new Product(0, "Mouse", Color.YELLOW, BigDecimal.valueOf(840));
+        blackDogFor500 = new Product(0, "Dog", Color.BLACK, BigDecimal.valueOf(500));
+        blackCatFor1001 = new Product(0, "Cat Black", Color.BLACK, BigDecimal.valueOf(1001));
+        blueCatFor1010 = new Product(0, "Cat", Color.BLUE, BigDecimal.valueOf(1010));
+        yellowMouseFor840 = new Product(0, "Mouse", Color.YELLOW, BigDecimal.valueOf(840));
         blackColorFilter = new ColorFilter(Color.BLACK);
+        whiteColorFilter = new ColorFilter(Color.WHITE);
         priceLessThanThousandFilter = new PriceLessThanFilter(BigDecimal.valueOf(1000));
-        atLeastFiveFilter = new AtLeastNOfFilter<>(2,blackColorFilter,priceLessThanThousandFilter);
+        atLeastTwoFilter = new AtLeastNOfFilter<>(2, blackColorFilter, priceLessThanThousandFilter);
 
-    }
+        products.add(blackDogFor500);
+        products.add(blueCatFor1010);
+        products.add(yellowMouseFor840);
+        products.add(blackCatFor1001);
 
-    @Test
-    public void select_selectWithBlackFilter_oneItemResultSuccess() {
-        List<Product> products = new ArrayList<>();
-        products.add(blackDog);
-        products.add(blueCat);
-        products.add(yellowMouse);
         try {
```

I think you can just let it fail